aardvark-node/src/node.rs

```diff
@@ -80,12 +80,12 @@ impl Node {
 
             inner
                 .network
-                .get_or_init(|| async {
```

Comment:
The reason for using get_or_init() was so that other places would wait for the network to be ready, e.g.  If the app calls `subscribe()` before the network is ready. Not sure if it's still needed since you also removed the async `get_or_init()` in methods that use the `network` but I think this introduces a race condition.

Reply:
Yes, you're right. When we're subscribing we want async access to the network instance. For the initialisation though I think we're fine with just using `set` as internally that uses the same semaphore and it's the only point where we actually set that value.